Index: arts/soundserver/artswrapper.c
===================================================================
--- arts/soundserver/artswrapper.c	(.../tags/arts/1.5.3)	(revision 549703)
+++ arts/soundserver/artswrapper.c	(.../branches/arts/1.5)	(revision 549703)
@@ -95,6 +95,10 @@
 #else
 		setreuid(-1, getuid());
 #endif
+		if (geteuid() != getuid()) {
+			perror("setuid()");
+			return 2;
+		}
 	}
 
 	if(argc == 0)
Index: arts/soundserver/crashhandler.cc
===================================================================
--- arts/soundserver/crashhandler.cc	(.../tags/arts/1.5.3)	(revision 549703)
+++ arts/soundserver/crashhandler.cc	(.../branches/arts/1.5)	(revision 549703)
@@ -196,7 +196,12 @@
           argv[i++] = NULL;
 
           setgid(getgid());
-          setuid(getuid());
+          if (getuid() != geteuid())
+            setuid(getuid());
+          if (getuid() != geteuid()) {
+	    perror("setuid()");
+            exit(255);
+          }
 
           execvp(crashApp, argv);
 
Index: arts/flow/gsl/gslmagic.c
===================================================================
--- arts/flow/gsl/gslmagic.c	(.../tags/arts/1.5.3)	(revision 549703)
+++ arts/flow/gsl/gslmagic.c	(.../branches/arts/1.5)	(revision 549703)
@@ -616,7 +616,7 @@
     return FALSE;
 
   do
-    ret = fstat (bfile->fd, &buf) < 0;
+    ret = fstat (bfile->fd, &buf);
   while (ret < 0 && errno == EINTR);
   if (ret < 0)
     {

Property changes on: arts
___________________________________________________________________
Name: svn:externals
   + libltdl https://svn.kde.org/home/kde/branches/KDE/3.5/kdelibs/libltdl
admin https://svn.kde.org/home/kde/branches/KDE/3.5/kde-common/admin


