--- arts-1.1.4/configure.in.orig	Wed Sep 10 03:53:29 2003
+++ arts-1.1.4/configure.in	Tue Mar 16 01:36:12 2004
@@ -150,6 +150,10 @@
 dnl aRts specific configure tests
 dnl
 dnl Check for libaudio (which can be used to get network audio support)
+AC_ARG_WITH(nas, [  --with-nas            enable network audio support],
+[arts_with_nas=$withval])
+
+if test "$arts_with_nas" = "yes"; then
 AC_DEFUN([AC_CHECK_LIBAUDIONAS],
 [
   ac_ldflags_save="$LDFLAGS"
@@ -177,6 +181,10 @@
   CPPFLAGS="$ac_CPPFLAGS_save"
   LDFLAGS="$ac_ldflags_save"
 ])
+AC_CHECK_LIBAUDIONAS
+fi
+
+AC_SUBST(LIBAUDIONAS)
 
 dnl libaudiofile is used for loading wave files
 AC_DEFUN([AC_CHECK_LIBAUDIOFILE],
@@ -428,7 +436,6 @@
 
 dnl call the tests
 AC_CHECK_LIBAUDIOIO
-AC_CHECK_LIBAUDIONAS
 AC_CHECK_LIBAUDIOFILE
 AC_CHECK_SGILIBAUDIO
 AC_CHECK_REALTIME_SCHED
