Index: tests/testifacerepo.cc
===================================================================
--- tests/testifacerepo.cc	(.../tags/arts/1.5.4/arts)	(revision 582125)
+++ tests/testifacerepo.cc	(.../branches/arts/1.5/arts)	(revision 582125)
@@ -21,7 +21,7 @@
     */
 
 #include <algorithm>
-#include "core.h"
+#include "common.h"
 #include "test.h"
 
 using namespace Arts;
Index: tests/testanyref.cc
===================================================================
--- tests/testanyref.cc	(.../tags/arts/1.5.4/arts)	(revision 582125)
+++ tests/testanyref.cc	(.../branches/arts/1.5/arts)	(revision 582125)
@@ -20,8 +20,7 @@
 
     */
 
-#include "anyref.h"
-#include "core.h"
+#include "common.h"
 #include "test.h"
 
 using namespace Arts;
Index: flow/audioioalsa.cc
===================================================================
--- flow/audioioalsa.cc	(.../tags/arts/1.5.4/arts)	(revision 582125)
+++ flow/audioioalsa.cc	(.../branches/arts/1.5/arts)	(revision 582125)
@@ -114,7 +114,11 @@
 	 */
     m_card = snd_defaults_pcm_card();  //!! need interface !!
     m_device = snd_defaults_pcm_device(); //!!
+#ifdef WORDS_BIGENDIAN
+    m_format = SND_PCM_SFMT_S16_BE;
+#else
     m_format = SND_PCM_SFMT_S16_LE;
+#endif
     m_bufferMode = block;  //block/stream (stream mode doesn't work yet)
 
 	if(m_card >= 0) {
Index: examples/referenceinfo.cc
===================================================================
--- examples/referenceinfo.cc	(.../tags/arts/1.5.4/arts)	(revision 582125)
+++ examples/referenceinfo.cc	(.../branches/arts/1.5/arts)	(revision 582125)
@@ -23,7 +23,7 @@
 
     */
 
-#include "core.h"
+#include "common.h"
 #include <stdio.h>
 
 using namespace Arts;
Index: examples/x11commtest.cc
===================================================================
--- examples/x11commtest.cc	(.../tags/arts/1.5.4/arts)	(revision 582125)
+++ examples/x11commtest.cc	(.../branches/arts/1.5/arts)	(revision 582125)
@@ -25,7 +25,7 @@
     */
 
 #include <iostream>
-#include "core.h"
+#include "common.h"
 
 using namespace std;
 using namespace Arts;
Index: examples/helloserver.cc
===================================================================
--- examples/helloserver.cc	(.../tags/arts/1.5.4/arts)	(revision 582125)
+++ examples/helloserver.cc	(.../branches/arts/1.5/arts)	(revision 582125)
@@ -23,7 +23,7 @@
 
     */
 
-#include "core.h"
+#include "common.h"
 #include "hello_impl.h"
 
 #include <stdio.h>
Index: Makefile.am.in
===================================================================
--- Makefile.am.in	(.../tags/arts/1.5.4/arts)	(revision 582125)
+++ Makefile.am.in	(.../branches/arts/1.5/arts)	(revision 582125)
@@ -21,7 +21,7 @@
 
 $(top_srcdir)/acinclude.m4: $(top_srcdir)/libltdl/ltdl.m4
 
-AUTOMAKE_OPTIONS = foreign 1.5
+AUTOMAKE_OPTIONS = foreign 1.6.1
 COMPILE_FIRST = libltdl mcop mcopidl flow mcop_mt soundserver artsc examples tests doc 
 EXTRA_DIST = admin
 

Property changes on: .
___________________________________________________________________
Name: svn:externals
   + libltdl https://svn.kde.org/home/kde/branches/KDE/3.5/kdelibs/libltdl
admin https://svn.kde.org/home/kde/branches/KDE/3.5/kde-common/admin


